### PR TITLE
Change preview source

### PR DESCRIPTION
**Pull request by Aradon267**

### CODE REVIEW

**Comment by YechiamW3** [comment]
@Aradon267 please that the origin of this PR is `supabase-migration`.

File: cv_next/app/actions/cvs/getBlurCv.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 
 export async function getBlurredCv(cvLink: string): Promise<string> {
-  const resp = await fetch(cvLink);
+  const resp = await fetch(cvLink, {
+    next: { revalidate: Definitions.FETCH_WAIT_TIME },
+  });
   const arrayBuffer = await resp.arrayBuffer();
   const buffer = Buffer.from(arrayBuffer);
-  const { base64 } = await getPlaiceholder(buffer, { size: 10 });
+  const { base64 } = await getPlaiceholder(buffer, { size: 15 });
```
Comment (YechiamW3) [inline cv_next/app/actions/cvs/getBlurCv.ts:11]: Use Definitions for this number.

File: cv_next/app/actions/cvs/preview.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    const { data, error } = await SupabaseHelper.getSupabaseInstance()
+      .storage.from("cvs")
+      .upload(fileName, blob, {
+        cacheControl: "3600",
+        upsert: true, // Replace if exists
+      });
+
+    if (error) {
+      MyLogger.logInfo("Upload error:", error);
```
Comment (YechiamW3) [inline cv_next/app/actions/cvs/preview.ts:34]: Same comment as in other PR about error logging.

File: cv_next/app/actions/cvs/preview.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    } else {
+      MyLogger.logInfo("File uploaded successfully:", data);
```
Comment (YechiamW3) [inline cv_next/app/actions/cvs/preview.ts:36]: Note that this will clutter our logs. Either `logDebug`, or add a TODO to delete this.

File: cv_next/app/actions/cvs/preview.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  }
+}
+
+export async function getImageURL(cvId: string): Promise<string | null> {
+  let data = SupabaseHelper.getSupabaseInstance()
+    .storage.from("cvs")
+    .getPublicUrl(cvId + ".png").data.publicUrl;
```
Comment (YechiamW3) [inline cv_next/app/actions/cvs/preview.ts:44]: Potential need for optional chaining.

File: cv_next/app/actions/cvs/preview.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+export async function getUserName(userId: string): Promise<string | null> {
+  const { data: user, error } = await SupabaseHelper.getSupabaseInstance()
+    .from("profiles")
+    .select("*")
+    .eq("id", userId)
+    .single();
+  if (error) {
+    return null;
+  }
+  return user.full_name;
```
Comment (YechiamW3) [inline cv_next/app/actions/cvs/preview.ts:57]:
```suggestion
  if (error) {
    //log..
  }
  return user?.full_name ?? null;
```

File: cv_next/app/feed/components/CVItemRSC.tsx [hunk 6]
```diff
@@ -1,61 +1,117 @@
...
+import Definitions from "@/lib/definitions";
 
-const getBlur = cache(async (imageURL: string) => {
-  return await getBlurredCv(imageURL);
-});
 interface CVCardProps {
   cv: CvModel;
 }
 
+const getBlur = cache(async (imageURL: string) => {
```
Comment (YechiamW3) [inline cv_next/app/feed/components/CVItemRSC.tsx:19]: All of these are obsolete if the function is Client component.
See: https://react.dev/reference/react/cache
TL;DR: use `useMemo`.

File: cv_next/app/feed/components/CVItemRSC.tsx [hunk 7]
```diff
@@ -1,61 +1,117 @@
 "use client";
+import { getBlurredCv } from "@/app/actions/cvs/getBlurCv";
 import Image from "next/image";
 import Link from "next/link";
-import generateImageUrl from "@/helpers/imageURLHelper";
 import Categories from "@/types/models/categories";
-import { useEffect, useState, cache } from "react";
-import { getBlurredCv } from "@/app/actions/cvs/getBlurCv";
+import { useState, useEffect, cache } from "react";
```
Comment (YechiamW3) [inline cv_next/app/feed/components/CVItemRSC.tsx:6]: Should probably rename this file/merge with `CVItem` if we don't RSC anymore here.

File: cv_next/app/feed/components/CVItemRSC.tsx [hunk 8]
```diff
@@ -1,61 +1,117 @@
...
+    };
+    const getBlurCv = async () => {
+      const base64 = await getBlur(getGoogleImageUrl(cv.document_link));
       setBase64Data(base64);
-    })();
-  }, [imageUrl]);
+    };
+    getBlurCv();
+    getAuthorName();
+    revalidateImage(); // Immediately invoke revalidateImage
```
Comment (YechiamW3) [inline cv_next/app/feed/components/CVItemRSC.tsx:60]: FYI, if you don't `await` on each of these functions you're calling them all asynchronously, i.e. they don't depend on each other. Just making sure because I don't understand the comment.

File: cv_next/app/feed/components/CVItemRSC.tsx [hunk 9]
```diff
@@ -1,61 +1,117 @@
...
+          height={500 * 1.4142}
+          className="w-full rounded-lg p-2"
+          src={realURL}
+          placeholder="blur"
+          blurDataURL={base64Data}
+          alt="CV Preview"
+          priority
+        />
+      ) : (
+        <div style={{ width: "380px", height: `400px`, overflow: "hidden" }}>
```
Comment (YechiamW3) [inline cv_next/app/feed/components/CVItemRSC.tsx:86]: Why change width and height via `style` when there are specific attributes for that in the `Image` component?

File: cv_next/helpers/blobHelper.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+async function calculateHash(blob: Blob): Promise<string> {
```
Comment (YechiamW3) [inline cv_next/helpers/blobHelper.ts:1]: Please add JSDoc for these functions.

File: cv_next/helpers/imageURLHelper.ts [hunk 11]
```diff
@@ -1,16 +1,16 @@
...
  * @return {string | null} The extracted ID or null if no ID is found.
  */
-function getIdFromLink(link: string): string | null {
-  const idPattern = /\/d\/([a-zA-Z0-9_-]+)/
-  const match = link.match(idPattern)
-  return match?.[1] ?? null
+export function getIdFromLink(link: string): string | null {
+  const idPattern = /\/d\/([a-zA-Z0-9_-]+)/;
+  const match = link.match(idPattern);
+  return match?.[1] ?? null ?? null;
```
Comment (YechiamW3) [inline cv_next/helpers/imageURLHelper.ts:13]: `?? null ?? null`?

File: cv_next/app/actions/cvs/preview.ts (hunk 2 above)
Comment (Aradon267) [reply cv_next/app/actions/cvs/preview.ts:34]: Avner started working with winaton in the cv page with supabase
I'll add it to this branch as well and replace all

File: cv_next/app/feed/components/CVItemRSC.tsx (hunk 8 above)
Comment (Aradon267) [reply cv_next/app/feed/components/CVItemRSC.tsx:60]: Technically yeah this comment is redundant but these should really depend on each other(maybe only the getblurcv and the revalidate because they both fetch the entry from google but they both also cache it so not really sure if it matters which one runs first)

File: cv_next/app/feed/components/CVItemRSC.tsx (hunk 9 above)
Comment (Aradon267) [reply cv_next/app/feed/components/CVItemRSC.tsx:86]: This is mainly for the initial load(a default valid value must be given to the Image for it to load properly this makes sure that the default data spreads to the entire space instead of only taking what it needs)

File: cv_next/app/actions/cvs/preview.ts (hunk 4 above)
Comment (Aradon267) [reply cv_next/app/actions/cvs/preview.ts:44]: Not sure as this is called only after we revalidate the image(either reupload it to supabase or it was there so nothing happened)

File: cv_next/app/actions/cvs/preview.ts (hunk 2 above)
Comment (YechiamW3) [reply cv_next/app/actions/cvs/preview.ts]: See comment on other PR

File: cv_next/app/feed/components/CVItemRSC.tsx (hunk 8 above)
Comment (YechiamW3) [reply cv_next/app/feed/components/CVItemRSC.tsx]: Well, if you want to make sure e.g. getBlurCv is ran before revalidateImage you should `await getBlurCv()`..

File: cv_next/app/actions/cvs/preview.ts (hunk 2 above)
Comment (Aradon267) [reply cv_next/app/actions/cvs/preview.ts]: Resolving on other PR so we can focus this here
I'll create a new branch from dev
As this is the last thread for this PR, should we wait for the logger, merge it, and fix on each branch or merge these and fix in the logger branch?

Comment (YechiamW3) [reply cv_next/app/actions/cvs/preview.ts]: Merge this branch and fix on the logger branch
